pkg/providers/vault/client.go

```
@@ -48,11 +60,139 @@ func NewSecretClient(config SecretConfig) (pkg.SecretClient, error) {
 		config.retryWaitPeriodTime = retryTimeDuration
 	}
 
-	return Client{
+	secretClient := Client{
 		HttpConfig: config,
 		HttpCaller: httpClient,
-	}, nil
+	}
+
+	tokenStr := config.Authentication.AuthToken
```

I think it is an error for config.Authentication.AuthToken to be blank.  At the minimum I would invert the if statement and exit early if AuthToken is empty.  But I'd rather outright fail earlier in the flow if this parameter is missing.

---

not sure this is always the case. Prior to this, we do not event check whether the token string is empty or not

---

after discussion, can make AuthToken as a required input hence forcing error if not for the caller